app/src/ui/app-theme.tsx

```diff
@@ -44,19 +58,128 @@ export class AppTheme extends React.PureComponent<IAppThemeProps> {
+
+        --secondary-button-hover-border-width: 2px;
+
+        --tab-bar-box-shadow: none;
+
+        --diff-add-border: 1px solid green;
+        --diff-delete-border: 1px solid crimson;
+        --tab-bar-hover-border: 2px solid ${border} !important;
+        --tab-bar-item-border: 2px solid ${background};
+        --foldout-border: 1px solid ${border};
+        --horizontal-bar-active-color: ${activeItem};
+        --horizontal-bar-active-text-color: ${activeText};
+    `
+
+    styles.appendChild(
+      document.createTextNode(
+        `body.theme-custom {
+            --background-color: ${background};
+            --box-background-color: ${background};
+            --box-alt-background-color: ${background};
+            --box-alt-text-color: ${activeText};
+
+            --diff-hunk-gutter-background-color: ${background};
+            --diff-text-color: ${text};
+            --diff-line-number-color: ${text};
+            --diff-gutter-background-color: ${background};
+            --diff-hunk-background-color: ${background};
+            --diff-empty-row-background-color: ${secondaryBackgroundColor};
+
+            --box-border-color: ${border};
+            --diff-border-color: ${border};
+
+            --box-selected-background-color: ${secondaryActiveColor};
+            --box-selected-text-color: ${activeText};
+
+            --box-selected-active-background-color: ${activeItem};
+            --box-selected-active-text-color: ${activeText};
+
+            --button-background: ${activeItem};
+            --button-text-color: ${activeText};
+            --secondary-button-background: ${background};
+            --secondary-button-text-color: ${text};
+            --button-hover-background: ${secondaryActiveColor};
+            --secondary-button-hover-background: ${secondaryBackgroundColor};
+            --app-menu-button-hover-background-color: ${secondaryBackgroundColor};
+            --toolbar-button-focus-background-color: ${secondaryBackgroundColor};
+            --toolbar-button-hover-background-color: ${secondaryBackgroundColor};
+            --toolbar-button-active-border-color: ${border};
+            --tab-bar-hover-background-color: ${secondaryBackgroundColor};
+
+            --text-color: ${text};
+            --text-secondary-color: ${text};
+            --toolbar-background-color: ${background};
+            --toolbar-button-secondary-color: ${text};
+
+            --list-item-hover-background-color: ${secondaryActiveColor};
+            --list-item-hover-text-color: ${activeText};
+
+            --box-placeholder-color: ${text};
+            --tab-bar-active-color: ${activeItem};
 
-    body.classList.add(newThemeClassName)
+            ${highContrastSpecific}
+          }`
+      )
+    )
+    body.appendChild(styles)
   }
 
   private clearThemes() {
     const body = document.body
 
-    for (const className of body.classList) {
+    const classList = [...body.classList]
+    for (const className of classList) {
```

Comment:
This is one of those .. I don't know why it didn't work ... it only worked for our app before because we only had `-dark` or `-light`.. now we have (`-dark` or `-light`) and `-custom` (two theme classes at a time) and I found that when switching between dark and light and high contrast that it would only remove one of those classes..

My theory is that `body.classList` is not a regular array (it some html dom construct) and the iterator didn't work like a normal (again no idea why) array.  Using the `[...body.classList]` forces it into a "normal" array... and therefore, it will actually clear all the themes.

Reply:
🤯